### DISCUSSION TITLE
Default to "Show all details?"

### DISCUSSION THREAD

**Discussion by Joecey** [pytest-dev/pytest-html#601] [Q&A] [answered]
Hi all, 

I was wondering if there is a way when creating/opening a report to have "Show all details" by default? Essentially, I am looking for the opposite of "render_collapsed = True." Currently, if I set this to false, only failed Test Cases expand while passed TCs are collapsed (I want all TCs to be expanded regardless of state)

Thanks in advance!

**Comment by BeyondEvil** [comment, marked as answer]
That's a good point. This might actually already be supported in 4.0.0rc0.

If not, I'll see if there's an easy way of adding support for it.

Perhaps we could add support for something like "collapsed=none" to have them all exapanded regardless of state.

**Reply by Joecey** [reply]
Not a big deal if it's not available yet. While it would be a nice add-on, it's more of a time saver for my team (we want to automate converting the HTML to PDF, and ideally we want all details to be shown by default). 

My guess would have been to add the following in result.py 
`if self.config.getini("render_collapsed") == False:`

and setting `tr_class = None` but I don't think it's as simple as that. I might create a fork and take a deeper dive if I'm free in the evening

**Reply by BeyondEvil** [reply]
> Not a big deal if it's not available yet. While it would be a nice add-on, it's more of a time saver for my team (we want to automate converting the HTML to PDF, and ideally we want all details to be shown by default).
> 
> My guess would have been to add the following in result.py `if self.config.getini("render_collapsed") == False:`
> 
> and setting `tr_class = None` but I don't think it's as simple as that. I might create a fork and take a deeper dive if I'm free in the evening

I'm happy to help with any contributions.

Do note however that v4.0.0 (that is in RC currently) is a _complete_ rewrite. So make sure you're working off the `next-gen` branch if you're going to contribute.

Do also note that there's a lot happening on that branch and nothing is considered stabled.

**Comment by BeyondEvil** [comment]
There's a PR for this functionality now.

Basically if you pass `?collapsed`, `?collapsed=` or `?collapsed=""` all we will be expanded - included "passed".

Conversely, passing `?collapsed=all` will.... collapse all. :)

I'll let you know when RC1 is out.

**Comment by BeyondEvil** [comment]
`4.0.0rc1` just released! Please try it out. 🙇
